core/engine/Cargo.toml

```diff
@@ -14,10 +14,15 @@ rust-version.workspace = true
 [features]
 profiler = ["boa_profiler/profiler"]
 deser = ["boa_interner/serde", "boa_ast/serde"]
-intl = [
+
+# Enables the `Intl` builtin object and bundles a default ICU4X data.
+intl = ["intl_core", "dep:boa_icu_provider"]
+
+# Enables only the `Intl` builtin implementation without any bundled ICU4X data provider.
+intl_core = [
```

Comment:
I think it would be better to keep the intl name for enabling intl functionality, and add name the flag that bakes the blob in the executable, something like `intl-blob`.

Calling it just intl to me sounds like we are activating all intl functionality, and intl_core sounds like you are enabling a sub part of the functionality.

Reply:
Will invert the names. About `intl-blob`, I think it would be better to pick a name that doesn't expose our internal representation of the ICU4X data, since we could migrate to other formats or even "baked" data in the future. Some other ideas:
- default_icu_data
- intl_data (preference)
- icu_intl

Reply:
`intl_data` seems good to me :)